modules/metrics/src/main/java/com/spotify/apollo/metrics/semantic/SemanticServiceMetrics.java

```diff
@@ -191,6 +197,31 @@ private CachedMeters metersForEndpoint(String endpoint) {
            Optional.empty();
   }
 
+  /**
+   * Checks for duration-threshold-metrics configuration options and sets up
+   * metrics that track how many requests meet a certain duration threshold goal.
+   *
+   * @param id
+   * @param endpoint formatted as METHOD:URI
+   * @return
+   */
+  private Optional<DurationThresholdTracker> requestDurationThresholdTracker(MetricId id, String endpoint) {
+    final String[] endpointComponents = endpoint.split(":");
```

Comment:
What happens when the endpoint doesn't contains `:` ?

Reply:
@tommyulfsparre I looked and it seems that endpoint always contains this, as it is passed from https://github.com/spotify/apollo/blob/master/modules/metrics/src/main/java/com/spotify/apollo/metrics/MetricsCollectingEndpointRunnableFactoryDecorator.java#L46 which calls `EndpointInfo.getName()` which always concats method and route: https://github.com/spotify/apollo/blob/725ba23c2dcc2ee0aecbf669e2166d76c4ce39cc/apollo-api-impl/src/main/java/com/spotify/apollo/dispatch/EndpointInfo.java#L60

It may be a better idea to refactor to pass down method and route separately. Happy to do that, if you agree.

Reply:
Nah, It fine as is. The other public method didn't have any documentation either and that code is unlikely to change.